pkg/odo/cli/url/delete.go

```
@@ -24,23 +25,34 @@ var (
 
 // URLDeleteOptions encapsulates the options for the odo url delete command
 type URLDeleteOptions struct {
-	localConfigInfo    *config.LocalConfigInfo
-	componentContext   string
+	*clicomponent.CommonPushOptions
 	urlName            string
 	urlForceDeleteFlag bool
-	*genericclioptions.Context
+	now                bool
 }
 
 // NewURLDeleteOptions creates a new URLDeleteOptions instance
 func NewURLDeleteOptions() *URLDeleteOptions {
-	return &URLDeleteOptions{}
+	return &URLDeleteOptions{CommonPushOptions: clicomponent.NewCommonPushOptions()}
 }
 
 // Complete completes URLDeleteOptions after they've been Deleted
 func (o *URLDeleteOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	o.Context = genericclioptions.NewContext(cmd)
+	if o.now {
+		o.Context = genericclioptions.NewContextCreatingAppIfNeeded(cmd)
+	} else {
+		o.Context = genericclioptions.NewContext(cmd)
+	}
 	o.urlName = args[0]
-	o.localConfigInfo, err = config.NewLocalConfigInfo(o.componentContext)
+	err = o.InitConfigFromContext()
```

this error is never checked

---

@kadel fixed